frontend/javascripts/dashboard/dataset/dataset_cache_provider.js

```diff
@@ -89,8 +91,34 @@ export default function DatasetCacheProvider({ children }: { children: Node }) {
       setIsLoading(false);
     }
   }
+
+  async function updateDataset(datasetId: APIDatasetId) {
+    if (isLoading) return;
```

Comment:
Could you also add this line to the `fetchDatasets` method above. Otherwise, I'm worried about race conditions where this method is invoked first, then `fetchDatasets` is called and the results are applied in undefined order.

Reply:
I noticed that this would lead to the abortion of fetching the datasets after triggering the check in `checkDatasets`. Please see if my solution is fine :slightly_smiling_face: